### PR TITLE
Fix typos in docs

### PR DESCRIPTION
**Pull request by Overcastan**
This PR fixes small typos in Miden VM documentation.

### CODE REVIEW

File: docs/src/design/stack/u32_ops.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Most operations described below require some number of 16-bit range checks (i.e.
 To perform these range checks, the prover puts the values to be range-checked into helper registers $h_0, ..., h_3$, and then divides the range checker bus column $b_{range}$ by a randomized product of these values. This operation is enforced via the following constraint:
 
 >$$
-b_{range}' \cdot (\alpha_0 + h_0) \cdot (\alpha_0 + h_1) \cdot (\alpha_0 + h_2) \cdot (\alpha_0 + h_3) = b_{range} \text{ | degree} = 5
+b_{range}' \cdot (\alpha_0 + h_0) \cdot (\alpha_1 + h_1) \cdot (\alpha_2 + h_2) \cdot (\alpha_3 + h_3) = b_{range} \text{ | degree} = 5
```
Comment (bobbinth) [inline docs/src/design/stack/u32_ops.md:10]: This was actually correct here.

File: docs/src/design/decoder/constraints.md [hunk 2]
```diff
@@ -310,12 +310,12 @@ In addition to the above transition constraint, we also need to set the followin
...
-The `in_span` column (denoted as $sp$) is used to identify rows which execute non-control flow operations. The values in this column are set as follows:
+The `is_span` column (denoted as $sp$) is used to identify rows which execute non-control flow operations. The values in this column are set as follows:
 
-* Executing a `SPAN` operation sets the value of `in_span` column to $1$.
+* Executing a `SPAN` operation sets the value of `is_span` column to $1$.
 * The value remains $1$ until the `END` operation is executed.
-* If `RESPAN` operation is executed between `SPAN` and `END` operations, in the row at which `RESPAN` operation is executed `in_span` is set to $0$. It is then reset to $1$ in the following row.
-* In all other cases, value in the `in_span` column should be $0$.
+* If `RESPAN` operation is executed between `SPAN` and `END` operations, in the row at which `RESPAN` operation is executed `is_span` is set to $0$. It is then reset to $1$ in the following row.
+* In all other cases, value in the `is_span` column should be $0$.
```
Comment (bobbinth) [inline docs/src/design/decoder/constraints.md:318]: Let's keep it as `in_span` here - that's how we also have it in the code.

**Review by bobbinth — CHANGES_REQUESTED** [review]
Thank you! I left a couple of small comments inline.

File: docs/src/design/decoder/main.md [hunk 3]
```diff
@@ -287,9 +287,9 @@ In the above diagram, `blk` is the ID of the *split* block which is about to be
 When the VM executes a `SPLIT` operation, it does the following:
 
 1. Adds a tuple `(blk, prnt, 0)` to the block stack table.
-2. Pops the stack and:
-   a. If the popped value is $1$, adds a tuple `(blk, true_branch_hash, 0, 0)` to the block hash table.
-   b. If the popped value is $0$, adds a tuple `(blk, false_branch_hash, 0, 0)` to the block hash table.
+2. Pops the stack and:\
+   a. If the popped value is $1$, adds a tuple `(blk, true_branch_hash, 0, 0)` to the block hash table.\
+   b. If the popped value is $0$, adds a tuple `(blk, false_branch_hash, 0, 0)` to the block hash table.\
```
Comment (bobbinth) [inline docs/src/design/decoder/main.md:292]: Why do we need this extra `\` here (and also in the similar paragraph on line 306?

Comment (Overcastan) [reply docs/src/design/decoder/main.md:292]: Without it this nested list is looking like this:
```
2. Pops the stack and: a. If the popped value is 1, adds a tuple (blk, true_branch_hash, 0, 0) to the block hash table. b. If 
   the popped value is 0, adds a tuple (blk, false_branch_hash, 0, 0) to the block hash table. c. If the popped value is 
   neither 1 nor 0, the execution fails.
```
Extra `\` here ads additional line break and created nested list:
```
2. Pops the stack and:
   a. If the popped value is 11, adds a tuple (blk, true_branch_hash, 0, 0) to the block hash table.
   b. If the popped value is 00, adds a tuple (blk, false_branch_hash, 0, 0) to the block hash table.
   c. If the popped value is neither 11 nor 00, the execution fails.
``` 
At least this is how it is displayed for me, I still need to check whether it is displayed correctly for others.

Comment (bobbinth) [reply docs/src/design/decoder/main.md:292]: Ah - makes sense!

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**